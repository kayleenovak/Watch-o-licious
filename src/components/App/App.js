import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Route } from 'react-router-dom'
import { MainPage } from '../../conatiners/MainPage/MainPage'
import { CardContainer } from '../../containers/CardContainer/CardContainer'
import

class App extends Component {

  componentDidMount() {
    const cakeUrl = ''
    const bakeUrl = ''
    const bakingUrl = ''
    this.props.fetchTvShows(cakeUrl)
    this.props.fetchTvShows(bakeUrl)
    this.props.fetchTvShows(bakingUrl)
  }

  render() {
    return (
      <div className="App">
        <Route exact to='/' component={ MainPage } />
        <Route exact to='/favorites' component={ CardContainer } />
        <Route exact to='/watched' component={ CardContainer } />
        <Route exact to='/watchlist' component={ CardContainer } />
        <Route component={ NoMatch } />
      </div>
    );
  }
}

export default App;
