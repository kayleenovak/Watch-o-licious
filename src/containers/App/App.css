.app {
  margin: 0;
  width: 100%;
  padding: 10px;
  position: relative;
  background: linear-gradient(
      rgba(166, 212, 211, 0.41), 
      rgba(166, 212, 211, 0.41)
    ),
    url('/background.png');
}
