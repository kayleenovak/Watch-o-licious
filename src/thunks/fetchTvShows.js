import { isLoading, hasErrored, dataFetchSuccess } from '../actions/index.js'
import { tvShowsCleaner } from '../cleaners/tvShowsCleaner.js'

export const fetchTvShows = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const tvShows = await response.json()
      const cleanedTvShows = await tvShowsCleaner(tvShows)
      dispatch(dataFetchSuccess(cleanedTvShows))
    } catch (error) {
      dispatch(hasErrored(true))
    }
  }
}