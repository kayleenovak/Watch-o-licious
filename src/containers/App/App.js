import React, { Component } from 'react';
import './App.css';
import { Route, Switch, withRouter } from 'react-router-dom'
import MainPage from '../../containers/MainPage/MainPage'
import { fetchTvShows } from '../../thunks/fetchTvShows.js'
import { connect } from 'react-redux'
import TvShowModal from '../TvShowModal/TvShowModal.js'
import CardContainer from '../CardContainer/CardContainer'
import { Header } from '../../components/Header/Header.js'
import { NoMatch } from '../../components/NoMatch/NoMatch.js'
import PropTypes from 'prop-types'

export class App extends Component {

  async componentDidMount() {
    await this.props.fetchTvShows()
  }

  render() {
    return (
      <div className="app">
          <Header />
          <Switch>
            <Route exact path='/' component={ MainPage } />
            <Route exact path='/tvshow/:id' render={({ match }) => {
              const { id } = match.params
              return <TvShowModal id={ id } />
              }} 
            />
            <Route exact path='/favorites' component={ CardContainer } />
            <Route exact path='/watched' component={ CardContainer } />
            <Route exact path='/watchlist' component={ CardContainer } />
            <Route component={NoMatch} />
          </Switch>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchTvShows: (url) => dispatch(fetchTvShows(url))
})

export default withRouter(connect(null, mapDispatchToProps)(App))

App.propTypes = {
  fetchTvShows: PropTypes.func.isRequired
}

