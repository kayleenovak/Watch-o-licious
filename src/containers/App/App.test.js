import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps } from './App';
import { Route } from 'react-router-dom'
import { shallow } from 'enzyme'
import { fetchTvShows } from '../../thunks/fetchTvShows.js'

jest.mock('../../thunks/fetchTvShows.js')

describe('App', () => {
  let wrapper
  let mockFetchTvShows

  beforeEach(() => {
    mockFetchTvShows = jest.fn()
    wrapper = shallow(<App fetchTvShows={ mockFetchTvShows } />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should render 6 routes', () => {

    expect(wrapper.find(Route).length).toEqual(6)
  })

  it('should fire fetchTvShows on componentDidMount', () => {

    wrapper.instance().componentDidMount()

    expect(mockFetchTvShows).toHaveBeenCalled()
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn()
      const mockUrl = 'www.thisurl.com'
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.fetchTvShows(mockUrl)

      expect(mockDispatch).toHaveBeenCalledWith(fetchTvShows(mockUrl))
    })
  })
})
