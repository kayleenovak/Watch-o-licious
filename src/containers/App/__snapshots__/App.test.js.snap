// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchTvShows={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            exact={true}
            path="/tvshow/:id"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/watched"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/watchlist"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              exact={true}
              path="/tvshow/:id"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/watched"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/watchlist"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/tvshow/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/watched",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/watchlist",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              exact={true}
              path="/tvshow/:id"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/watched"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/watchlist"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                exact={true}
                path="/tvshow/:id"
                render={[Function]}
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/watched"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/watchlist"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/tvshow/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/watched",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/watchlist",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
